pilot/pkg/bootstrap/certcontroller.go

```diff
@@ -135,7 +135,14 @@ func (s *Server) initDNSCerts(hostname, customHost, namespace string) error {
 
 	var certChain, keyPEM []byte
 	var err error
-	if features.PilotCertProvider.Get() == KubernetesCAProvider {
+
+	k8sDefaultSignerSupported, k8sVersion := kubeDefaultSignerSupported(s.kubeClient)
```

Comment:
I am not sure checking the version is what we need/want here. In reality, its not based (just) on k8s versions, but which api version we use. If we use the v1 api, even 1.19 would fail. If a user is on 1.22 and we still use the v1beta1 api, it will fail due to the API not existing.

So it seems like we actually only care about the API version; k8s version is just a proxy for that.

Reply:
The plan is to integrate a newer client library provided here- https://github.com/kubernetes/client-go/tree/master/util/certificate/csr, into chiron as part of this effort (in the next PR). This new client library always falls back to using v1beta1 whenever we use the Kubernetes legacy-signer - therefore, from the perspective of using legacy-signer, the code will remain unchanged. 
 My understanding as per https://github.com/istio/istio/issues/22161 is that from 1.22, v1Beta1 API will become unsupported and hence the newer client library will also fail. This PR will ensure that regardless of the client library chiron uses (current/new library), we never attempt to use the legacy-signer for DNS certificates depending on the K8s version.

Reply:
Got it, if we are using dual-api versions than this check may make more sense. Thanks